arch/src/main/java/knaufdan/android/arch/base/component/ComponentBindingAdapters.kt

```diff
@@ -18,97 +19,94 @@ private val parentComponents: WeakHashMap<ViewGroup, List<IComponent<*>>> = Weak
 
 @BindingAdapter(
     value = [
-        "components",
+        "component",
         "transition"
     ],
     requireAll = false
 )
-fun ViewGroup.bindComponents(
-    components: List<IComponent<*>>?,
+fun ViewGroup.bindComponent(
+    component: IComponent<*>?,
     transition: ViewTransition?
 ) {
-    if (components == null) {
-        return
-    }
-
-    if (hasComponents(components)) {
+    if (component == null) {
         return
     }
-    parentComponents[this] = components.toList()
 
-    removeAllViewsInLayout()
-
-    components.forEach { component ->
-        bindOneComponent(
-            component = component,
-            transition = transition
-        )
-    }
+    bindComponents(
+        components = listOf(component),
+        transition = transition
+    )
 }
 
 @BindingAdapter(
     value = [
-        "component",
+        "components",
         "transition"
     ],
     requireAll = false
 )
-fun ViewGroup.bindComponent(
-    component: IComponent<*>?,
+fun ViewGroup.bindComponents(
+    components: List<IComponent<*>>?,
     transition: ViewTransition?
 ) {
-    if (component == null) {
+    if (components == null) {
         return
     }
-    if (hasComponents(listOf(component))) {
+
+    val newComponents = components.toList()
+    val oldComponents = parentComponents[this] ?: emptyList()
+    if (oldComponents.contentDeepEquals(newComponents)) {
         return
     }
-    parentComponents[this] = listOf(component)
 
-    removeAllViewsInLayout()
+    if (oldComponents.isEmpty()) {
+        newComponents.forEach { component ->
+            bindOneComponent(
+                component = component,
+                transition = transition
+            )
+        }
+        parentComponents[this] = newComponents
+        return
+    }
```

Comment:
This is unnecessary, in case of empty old `if-block` in line 76 is executed and adds component at end.

Reply:
done.